scripts/gitfunctions.sh

```diff
@@ -2,43 +2,196 @@
 #
 # Convenient functions for use on Nuxeo projects version controlled under Git.
 #
-# (C) Copyright 2009-2013 Nuxeo SA (http://nuxeo.com/) and contributors.
+# (C) Copyright 2009-2016 Nuxeo SA (http://nuxeo.com/) and contributors.
 #
-# All rights reserved. This program and the accompanying materials
-# are made available under the terms of the GNU Lesser General Public License
-# (LGPL) version 2.1 which accompanies this distribution, and is available at
-# http://www.gnu.org/licenses/lgpl-2.1.html
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
 #
-# This library is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
-# Lesser General Public License for more details.
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
 #
 # Contributors:
-#   Julien Carsique
+#   Julien Carsique, Kevin Leturc
 #
-HERE=$(cd $(dirname $BASH_SOURCE); pwd -P)
+
+if [ -n "$BASH" ]; then
+  HERE=$(cd $(dirname $BASH_SOURCE); pwd -P)
+elif [ -n "ZSH_NAME" ]; then
+  HERE=$(cd $(dirname ${(%):-%N}); pwd -P)
+else
+  HERE=$(cd $(dirname $_); pwd -P)
+fi
 . $HERE/nxutils.sh
 
-# usage: "gitf [git instructions]"
 gitf() {
-  for dir in . *; do
-    if [ -e "$dir"/.git ]; then
+  # Parse options
+  local OPTIND=1 # getopts reset
+  local recurse=false
+  local quiet=false
+  local opt
+  while getopts "qrh?" opt; do
+    case "$opt" in
+    q) quiet=true
+      ;;
+    r) recurse=true
+      ;;
+    h|\?)
+      echo -e "Usage: gitf [-r] [-q] [-h|-?] <[--] Git instructions>
+Recursively executes the given Git command on the current and its direct children Git repositories.\n
+\t-h,-?\tShow this help message and exit.
+\t-q\tQuiet mode (default is verbose).
+\t-r\tThe recurse will continue on all children Git repositories.
+\t--\tOptional separator between the options and the instructions which may start with a dash."
+      return
+      ;;
+    esac
+  done
+  shift $((OPTIND-1))
+  [ "$1" = "--" ] && shift
+  local git_args=$@
+
+  function git_command {
+    local dir
+    if [ -e .git ]; then
+      dir=$(basename $PWD)
       echo "[$dir]"
-      (cd "$dir" ; git "$@")
+      [ "$quiet" != true ] && echo "\$> git $git_args"
```

Comment:
Understood. I'll look at it. However I need the variable being set to ensure that it will be local, determined and not contaminated from outside or contaminating outside (especially the `recurse` one).